### PR TITLE
GH-4527 use lowercase name for config class name and use full names for inner classes

### PR DESCRIPTION
**Pull request by hmottestad**
GitHub issue resolved: #4527 

Briefly describe the changes proposed in this PR:

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by hmottestad** [comment]
@abrokenjester I was looking at the new CONFIG class that you introduced and have some suggestions for changes.

I propose we use Config instead of CONFIG. I haven't looked into why you went with CONFIG to begin with, but I was thinking that it breaks with our current naming conventions and I wasn't sure why.

I also noticed that there are some new abbreviations that I think will be confusing to users. I propose using the class names for the classes that the configs apply to. So `Config.CustomGraphQueryInferencer` instead of `Config.Cgqi`. I see that it messes a bit with the IntelliJ autocomplete, so maybe we could go with a suffix of sorts?

**Comment by hmottestad** [comment]
Since the old config classes are probably widely in use by users I think it might be a good idea to not remove them in 5.0.0 and leave them around until 6.0.0 so that users have extra long time to migrate their code.

**Comment by hmottestad** [comment]
Not sure why the PR verify is failing because the xml-format plugin can't be found.

**Comment by abrokenjester** [comment]
> @abrokenjester I was looking at the new CONFIG class that you introduced and have some suggestions for changes.
> 
> I propose we use Config instead of CONFIG. I haven't looked into why you went with CONFIG to begin with, but I was thinking that it breaks with our current naming conventions and I wasn't sure why.

We discussed this when I put up the original change - see https://www.eclipse.org/lists/rdf4j-dev/msg02136.html. The consensus was that we'd stick with all capitals because although that breaks with our general naming conventions, it aligns it with the convention that we use for all other vocabulary classes. They're all named for their prefix name, in all caps.

> I also noticed that there are some new abbreviations that I think will be confusing to users. I propose using the class names for the classes that the configs apply to. So `Config.CustomGraphQueryInferencer` instead of `Config.Cgqi`. I see that it messes a bit with the IntelliJ autocomplete, so maybe we could go with a suffix of sorts?

Two things:

1. the abbreviations align with what is used in the actual URIs - so ideally if you want to change these abbreviations you'll have to adapt the actual URIs as well (and make sure that all templates are adapted, too);
2. I chose the abbreviations based on brevity, both from a Java point of view _and_ for the actual RDF vocabulary itself. I agree that some, like `Cgqi` and `Ca`, are not the most intuitive, so happy to discuss alternatives, but I'd rather not go with the full class name as that makes it all _very_ verbose.

**Comment by hmottestad** [comment]
Sorry for not following along when this was being discussed. 

The main issue with the abbreviations is that they are really hard for users to discover. They have to read the Java docs to understand that the config for the ElasticsearchStore is now in ess.

I really believe we need to help users with discoverability. Maybe we could add a constant to all the main classes that point to the relevant vocabulary. For instance ElasticsearchStore.CONFIG would be equal to CONFIG.ess, and that way we would still maintain the vocabulary in a single location without any duplication.

**Comment by abrokenjester** [comment]
> The main issue with the abbreviations is that they are really hard for users to discover. They have to read the Java docs to understand that the config for the ElasticsearchStore is now in ess.

YMMV, but I don't find that "hard to discover". The IDE warns if a class or field is deprecated when you use it, as soon as you hover over it the javadoc tells you where to look for the replacement. 

That said, anything we can do to make it even easier is welcome of course :) 

> I really believe we need to help users with discoverability. Maybe we could add a constant to all the main classes that point to the relevant vocabulary. For instance ElasticsearchStore.CONFIG would be equal to CONFIG.ess, and that way we would still maintain the vocabulary in a single location without any duplication.

I like this idea.

**Review by abrokenjester — CHANGES_REQUESTED** [review]

File: core/model-vocabulary/src/main/java/org/eclipse/rdf4j/model/vocabulary/Config.java [hunk 1]
```diff
@@ -82,9 +84,11 @@ public static final class Rep {
 	}
 
 	/**
-	 * HTTP Repository config
+	 * HTTPRepository config
+	 *
+	 * @see org.eclipse.rdf4j.repository.http.HTTPRepository
 	 */
-	public static final class Http {
+	public static final class HTTPRepository {
```
[the remark this answers is not in the document]
Comment (abrokenjester) [reply core/model-vocabulary/src/main/java/org/eclipse/rdf4j/model/vocabulary/Config.java:91]: Fwiw I originally also considered not having these internal classes at all, and just have _everything_ as part of the `CONFIG` vocabulary class directly. Worth reconsidering perhaps? Certainly easier from a "code autocompletion" perspective...

**Comment by hmottestad** [comment]
I think I'll just leave things as they are here and instead look into adding some way to link to the relevant vocabulary from within the different sails and repos. Hopefully it's as easy as introducing a constant.

I'll think about the abbreviations for a bit and see if I can come up with anything that is cleared without being too long.

**Pull request closed without merging**